components/steering_wheel/include/IO_Digital.h

```diff
@@ -0,0 +1,58 @@
+/**
+ *IO_Digital.h
+ *Digital IO Module Header
+ */
+
+#pragma once
+
+/************************************************
+ * Includes
+ * *********************************************/
+
+/************************************************
+ * Defines
+ * *********************************************/
+
+#define NUM_INPUTS 7
+
+/************************************************
+ * Typedefs
+ * *********************************************/
+
+enum INPUT
+{
+    SWITCH1 = 0U,
+    SWITCH2,
+    SWITCH4,
+    SWITCH5,
+    BTN_TOP_LEFT,
+    BTN_MIDDLE,
+    BTN_TOP_RIGHT,
+    COUNT_INPUT,
+};
+// static_assert(COUNT_INPUT == NUM_INPUTS);
+
+#if 8 >= NUM_INPUTS
+typedef unsigned char io_digital_t;
+#elif 16 >= NUM_INPUTS
+typedef unsigned short io_digital_t;
+#else
+typedef unsigned int io_digital_t;
+#endif
+
+typedef struct
+{
+    io_digital_t input;
```

Comment:
instead of the `io_digital_t` add one of these inside the `io_digital_S` struct
```
    struct
    {
        bool switch1:1;
        bool switch2:1;
        etc.
    } bit;
```
it's called a bitfield, compiler can optimize everything into a minimal amount of memory